examples/vrf_on_shared_vc/jsontest.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"fmt"
+
+	metal "github.com/equinix-labs/metal-go/metal/v1"
+)
+
+func main() {
+	metalGatewayJson := `{"id":"589784be-113c-4d62-a2c2-ca7bbf1d6646","state":"ready","created_at":"2023-09-01T14:58:25Z","updated_at":"2023-09-01T14:58:25Z","project":{"href":"/metal/v1/projects/fa38c326-5fbc-4dd1-89fb-602fc93db208"},"virtual_network":{"href":"/metal/v1/virtual-networks/a5bee2da-2d71-403b-a782-77c11ab3a265"},"ip_reservation":{"id":"40aa3b9e-bfe1-4a82-bd76-51bc48b577ee","address_family":4,"netmask":"255.255.255.0","created_at":"2023-09-01T14:58:24Z","details":null,"tags":[],"public":false,"cidr":24,"management":false,"manageable":true,"customdata":{},"bill":false,"state":"created","project":{"href":"/metal/v1/projects/fa38c326-5fbc-4dd1-89fb-602fc93db208"},"project_lite":{"href":"/metal/v1/projects/fa38c326-5fbc-4dd1-89fb-602fc93db208"},"created_by":{"href":"/metal/v1/users/2579ffd0-f5df-4343-a11e-b9097ff7cf52"},"vrf":{"href":"/metal/v1/vrfs/bfd5c58d-e81d-470c-9f65-d15ef6e8734c"},"metal_gateway":{"href":"/metal/v1/metal-gateways/589784be-113c-4d62-a2c2-ca7bbf1d6646"},"network":"10.0.0.0","address":"10.0.0.2","gateway":"10.0.0.1","href":"/metal/v1/ips/40aa3b9e-bfe1-4a82-bd76-51bc48b577ee","metro":{"id":"d3d6b29f-042d-43b7-b3ce-0bf53d5754ca","name":"Dallas","code":"da","country":"US"},"type":"vrf"},"created_by":{"href":"/metal/v1/users/2579ffd0-f5df-4343-a11e-b9097ff7cf52"},"vrf":{"href":"/metal/v1/vrfs/bfd5c58d-e81d-470c-9f65-d15ef6e8734c"},"href":"/metal/v1/metal-gateways/589784be-113c-4d62-a2c2-ca7bbf1d6646"}`
```

Comment:
I'm stuck on this...

I attempted to get some validation in place by breaking the `type` enums for IPReservation and VrfIpReservation into separate component schemas (see the changes in `spec/oas3.patched` in this PR).  This causes the Go generator to generate proper `enums` for those fields, so I would _expect_ that:
1. The VrfIpReservation JSON in this sample response does not validate as an IPReservation model **AND**
2.  Because of (1), the VrfMetalGateway JSON in this sample does not validate as a MetalGateway model

It appears that only (1) is true: this file demonstrates that `err` is nil when unmarshalling this JSON into _either_ gateway type, but `gateway.IpReservation` is `nil`.  Presumably it's nil because the JSON is not a valid `IPReservation`, but I would have expected that to propagate up.  Is there an option we can set to make that happen?

Reply:
Also: it's important to call out that the strategy I've tried here in order to make the response JSON match only one schema in the `oneOf` for gateways is exactly the issue mentioned in https://github.com/equinix-labs/metal-go/pull/134#issuecomment-1701582292

Reply:
After more digging, it looks like this is a bug in the Go generator.  The `UnmarshalJSON` functions throw away any errors that happen due to invalid fields: https://github.com/equinix-labs/metal-go/blob/50e6dc401de3dd2ffef6b4788d9bc5f78e7dd5bb/metal/v1/model_vrf_ip_reservation.go#L913-L915

Reply:
The generator bug has been fixed, and this now resolves to a single model.